Homework/2/Program.cs

```
@@ -0,0 +1,470 @@
+                Console.WriteLine("5 - Add user to end of list\n");
+                Console.WriteLine("6 - Add user to n'th position in list\n");
+                Console.WriteLine("7 - Remove user from start of list\n");
+                Console.WriteLine("8 - Remove user from end of list\n");
+                Console.WriteLine("9 - Remove n'th user\n");
+                Console.WriteLine("10 - Sort users by username (asc/dec)\n");
+                Console.WriteLine("11 - Log out\n");
+            }
+            else {
+                Console.WriteLine("Select option:\n\n");
+                Console.WriteLine("1 - Log in\n");
+                Console.WriteLine("2 - Register\n");
+            }
+
+            Console.WriteLine("12 - Close console\n");
+
+            OptionSelection(option: Console.ReadLine());
+        }
+
+        // Loading users from file to array - HW3  / User details should follow this format: username;password - in different line for each user.
+        // Things to consider: line validation - multiple ;, empty usernames/passwords // ; char should also be validated on user creation
+        static void LoadUsersFromFile() {
+            try
+            {
+                using (StreamReader reader = new StreamReader(pathToFile))
+                {
+                    while (!reader.EndOfStream)
+                    {
+                        string[] userCredentials = reader.ReadLine().Split(';');
+                        if (CheckIfUserExists(userCredentials[0]))
+                        {
+                            throw new DuplicateUserCredentialsException(userCredentials[0]);
+                        }
+                        else
+                        {
+                            AddUserToArrayAtIndex(userCredentials, 0);
+                        }
+                    }
+                }
+            }
+            catch (FileNotFoundException fileEx)
+            {
```

Never have an empty catch block.  Not even in a practice program or a quick program you throw together.  Don't catch the exception at all or, at least, print `fileEx`.

You can break this rule only if you really don't care if you get an exception or not.  In that case you should add a comment explaining this.  For example, in some languages (not C#) the "MakeDirectory" method throws an exception if the directory already exists...

```
try
{
  MakeDirectory("WorkDirectory");
}
catch (DirectoryAlreadyExistsException)
{
  // This is ok, we can continue if the directory already exists
}
```